otp-authentication/README.md

```diff
@@ -0,0 +1,123 @@
+# OTP Authentication
+
+This is an example of Time-based One-Time Password (TOTP) authentication, compatible with Google Authenticator. It is a form of two factor authentication (2FA). Although the example uses basic email-password authentication, the example can easily be applied to other authentication providers too.
+
+![two-factor-authentication-with-laravel-and-google-authenticator-3-638](https://user-images.githubusercontent.com/852069/28504159-1602e500-7012-11e7-8fb1-6951e0c84c71.jpg)
+
+## Registration and Authentication
+
+### Registration Flow
+
+- User registers using email and password
+- User registers for OTP authentication
+- User activates OTP authentication by entering an OTP token from their Authenticator
+
+### Authentication Flow
+
+- User signs in using email and password
+- If OTP is not activated for the user, a token is returned
+- If OTP is activated, a preAuthentication response is returned
+- The client can ask the user for an OTP token
+- The OTP token is validated and a token is returned
+
+## Setting up the project
+
+### Schema
+
+Use the `graphcool-cli` to create a new Graphcool project, based on the schema, and open the Console:
+```bash
+graphcool init --schema ./schema.graphql
+graphcool console
+```
+### Permissions
+
+> **Warning! I have not configured permissions for this example. You need to configure protection for sensitive fields from the `User` Type yourself!**
+
+### Permanent Authentication token
+
+Create a new Permanent Authentication Token on the 'Authentication' tab of the 'Project Settings' page.
+The PAT needs to be named 'authenticateUser'. This PAT will be used by the schema extension mutation.
+
+### Webtask
+
+The functionality for this example is implemented in an external webtask. This is due to a number of external NPM modules, not available for inline functions.
```

Comment:
`node-qr-image` and `thirty-two`. I couldn't find any viable alternatives too on https://tehsis.github.io/webtaskio-canirequire/.